services/pull/update.go

```diff
@@ -16,61 +16,67 @@ import (
 	user_model "code.gitea.io/gitea/models/user"
 	"code.gitea.io/gitea/modules/git"
 	"code.gitea.io/gitea/modules/log"
-	repo_module "code.gitea.io/gitea/modules/repository"
 )
 
 // Update updates pull request with base branch.
-func Update(ctx context.Context, pull *issues_model.PullRequest, doer *user_model.User, message string, rebase bool) error {
-	var (
-		pr    *issues_model.PullRequest
-		style repo_model.MergeStyle
-	)
+func Update(ctx context.Context, pr *issues_model.PullRequest, doer *user_model.User, message string, rebase bool) error {
+	if pr.Flow == issues_model.PullRequestFlowAGit {
+		// TODO: update of agit flow pull request's head branch is unsupported
+		return fmt.Errorf("update of agit flow pull request's head branch is unsupported")
+	}
 
-	pullWorkingPool.CheckIn(fmt.Sprint(pull.ID))
-	defer pullWorkingPool.CheckOut(fmt.Sprint(pull.ID))
+	pullWorkingPool.CheckIn(fmt.Sprint(pr.ID))
+	defer pullWorkingPool.CheckOut(fmt.Sprint(pr.ID))
 
-	if rebase {
-		pr = pull
-		style = repo_model.MergeStyleRebaseUpdate
-	} else {
-		// use merge functions but switch repo's and branch's
-		pr = &issues_model.PullRequest{
-			HeadRepoID: pull.BaseRepoID,
-			BaseRepoID: pull.HeadRepoID,
-			HeadBranch: pull.BaseBranch,
-			BaseBranch: pull.HeadBranch,
-		}
-		style = repo_model.MergeStyleMerge
+	diffCount, err := GetDiverging(ctx, pr)
+	if err != nil {
+		return err
+	} else if diffCount.Behind == 0 {
+		return fmt.Errorf("HeadBranch of PR %d is up to date", pr.Index)
```

Comment:
Is that really an error?
Wouldn't 
```suggestion
		return nil // head branch is up to date
```
suffice?

Reply:
This is unchanged and as far I can see this error is used in the UI.